.github/workflows/build.yaml

```diff
@@ -140,303 +144,56 @@ jobs:
-        read REF TOOLSET < .github/scripts/vcpkg_ref
-        cd ${{ env.VCPKG_ROOT }}
-        # REF can be empty, a commit hash or a pull request
-        [ -n "$REF" ] && git fetch
-        git checkout $REF || git pull origin $REF
-        echo "vcpkg_key=${{ hashFiles( './install-dependencies.sh' ) }}-${REF}" >> $GITHUB_OUTPUT
-        echo "toolset=${TOOLSET}" >> $GITHUB_OUTPUT
-      shell: bash
-    - name: Restore vcpkg and its artifacts.
-      uses: actions/cache@v3
-      with:
-        path: ${{ env.VCPKG_ROOT }}\installed
-        key: |
-          ${{ steps.prepare.outputs.vcpkg_key }}-${{ matrix.arch }}
-    - name: Installing dependencies
-      run: |
-        choco install innosetup
-        # Cache should have been populated by vcpkg-packages job
-        ./install-dependencies.sh vcpkg --triplet=${{ env.VCPKG_TARGET_TRIPLET }}
-      shell: bash
-    - name: Configure MSVC development console
-      uses: ilammy/msvc-dev-cmd@v1
-      with:
-        arch: ${{ matrix.arch }}
-        # Check https://github.com/actions/runner-images/blob/main/images/win/toolsets/toolset-2022.json for installed versions
-        toolset: ${{ steps.prepare.outputs.toolset }}
-    - name: Compiler
-      env:
-        VCPKG_ROOT: C:\vcpkg
-      run: |
-        mkdir $env:GITHUB_WORKSPACE\build
-        cd $env:GITHUB_WORKSPACE\build
-        cmake.exe -G "NMake Makefiles" .. -DVCPKG_TARGET_TRIPLET=${{ env.VCPKG_TARGET_TRIPLET }} -DCMAKE_TOOLCHAIN_FILE="${{ env.VCPKG_ROOT }}/scripts/buildsystems/vcpkg.cmake" -DCMAKE_BUILD_TYPE=${{ matrix.config }} -DOPTION_BUILD_WEBSITE_TOOLS=OFF -DOPTION_BUILD_TRANSLATIONS=ON -DOPTION_BUILD_TESTS=ON -DOPTION_ASAN=OFF -DOPTION_BUILD_CODECHECK=OFF -DOPTION_BUILD_WINSTATIC=ON -DOPTION_USE_GLBINDING=ON -DOPTION_FORCE_EMBEDDED_MINIZIP=ON
-        if ((Select-String -Quiet REVDETECT-BROKEN VERSION)) { exit 1 }
-        nmake
-        if ("${{ matrix.config }}" -Match "Release") {
-          strip -sv ./src/widelands.exe
-        }
-    - name: InnoSetup
-      run: |
-        cd $env:GITHUB_WORKSPACE
-        # Environment variables needed by our InnoSetup script
-        $env:PLATFORM = "${{ matrix.arch }}"
-        $env:CONFIGURATION = "${{ matrix.config }}"
-        $env:APPVEYOR_BUILD_FOLDER = $env:GITHUB_WORKSPACE
-        $env:APPVEYOR_BUILD_VERSION = "Widelands-${{ github.sha }}-${{ matrix.config }}-${{ matrix.arch }}"
-        ISCC.exe /o$env:GITHUB_WORKSPACE /fWidelands-${{ github.sha }}-msvc-${{ matrix.config }}-${{ matrix.arch }} $env:GITHUB_WORKSPACE\utils\win32\innosetup\Widelands.iss
-    - name: Uploading installer
-      uses: actions/upload-artifact@v3
-      with:
-        name: Widelands ${{ matrix.config }} ${{ matrix.arch }} Installer (MSVC)
-        path: ${{ github.workspace }}\Widelands-${{ github.sha }}-msvc-${{ matrix.config }}-${{ matrix.arch }}.exe
+    if: >
+      needs.clang_tidy.outputs.create_installer == 'true' ||
+      needs.clang_tidy.outputs.inst_windows_msvc == 'true'
```

Comment:
Actually is there anything that is not checked for installers? Maybe we'll have to rethink whether we really want to create installers for all PR workflow runs, because that means almost half of the compilation jobs are always run.

Reply:
Yes, there are README files, utils/, debian/, test/, (xdg/ for some), ...

Rethinking makes sense. But in a seperate pr.